### PR TITLE
review cluster article

### PR DESCRIPTION
**Pull request by garaud**

### CODE REVIEW

File: articles/cluster_stations.org [hunk 1]
```diff
@@ -0,0 +1,137 @@
+#+TITLE: Cluster bike sharing stations in french cities
+#+AUTHOR: Damien Garaud <damien.garaud@oslandia.com>, Raphaël Delhome <raphael.delhome@oslandia.com>
+
+[[http://oslandia.com/][Oslandia]] team is involved in a constant effort in geospatial data gathering and
+analysis. By taking advantage of the recent trend in public open data
+releasing, and after reading an inspiring work done by James Lawlor for [[https://towardsdatascience.com/usage-patterns-of-dublin-bikes-stations-484bdd9c5b9e][Dublin
+data]], we decided to evaluate the situation in our own living places. Here comes
+the open data portals for both cities: [[https://data.grandlyon.com/][Lyon]] and [[http://opendata.bordeaux.fr/][Bordeaux]].
+
+* What is the data looking like?
```
Comment (mhugo) [inline articles/cluster_stations.org]: data = plural

File: articles/cluster_stations.org [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+Both data set have been gathered with a cron plugged onto open data portals to
+get data every 5 minutes. The data of this study are between the 8th of July and
+the 26th of September, /i.e./ more than 11 weeks of data.
+
+It is important to consider there is no real standardization relatively to bike
+sharing open data sets: each city provide its own data set with its own
```
Comment (mhugo) [inline articles/cluster_stations.org]: provides

File: articles/cluster_stations.org [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+the data set as simple as possible and work on comparable situations, we have
+agregated the measurement to one-hour periods, and dropped the data gathered
+during week-end days. We consequently have dataframes of =N= rows and 24
+columns (a column being the bike percentage of availability at every stations,
+for a given hour of the day).
+
+We have worked with four clusters, regarding elbow method as well as cluster
+significance.
+
+As showed by figures below, the global pattern seems comparable between both
```
Comment (mhugo) [inline articles/cluster_stations.org]: shown

File: articles/cluster_stations.org [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+#+CAPTION: Bike availability patterns in Bordeaux
+#+NAME: fig:bordeaux_availability_pattern
+[[../image/bordeaux-pattern.png]]
+
+#+CAPTION: Bike availability patterns in Lyon
+#+NAME: fig:lyon_availability_pattern
+[[../image/lyon-pattern.png]]
+
+* Understand a city from its bike sharing stations
```
Comment (mhugo) [inline articles/cluster_stations.org:98]: Understanding ? (not sure)

File: articles/cluster_stations.org [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+allow us to draw interactive maps with our newly designed clusters.
+
+#+CAPTION: Bike sharing stations in Bordeaux
+#+NAME: fig:bordeaux_map_4_clusters
+[[../image/bordeaux-4-clusters.png]]
+
+The situation of both example cities is quite diverging here, as illustrated by
+the previous maps. On the first hand, we can see that the Bordeaux spatial
+repartition is not that clear following the clustering process. Either the
+cluster process gives unsignificant results, or the shared bike usage does not
```
Comment (mhugo) [inline articles/cluster_stations.org]: insignificant

File: articles/predict_bike_availability.org [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+* Predicting shared bike availability
+
+The first practical question to answer here is the prediction horizon. Here we
+will attempt to predict the bike availability after 30 minutes. It could be a
+typical problem for a user who wants to plan a local trip: will he find some
+bike at his prefered station, or should he walk to the next one? Or maybe
+should he look for an alternative transportation mode, as there is no bike in
+the near neighborhood?
+
+Let use two weeks of data for training (/i.e./ from 11/07/11 to 25/07/11) so as
```
Comment (mhugo) [inline articles/predict_bike_availability.org]: Let's

File: articles/predict_bike_availability.org [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+id, hours, minutes) and the station-related features (numbers of available
+bikes and stands).
+
+To do the hard prediction job, we use [[https://github.com/dmlc/xgboost][XGBoost]] (see doc [[http://xgboost.readthedocs.io/en/latest/][here]]), a distributed
+gradient boosting method that can undertake classification as well as
+regression processes. Here we are in the second case, as we want to estimate
+the value of a quantitative variable (the probability of finding an available
+bike at a given station, at a given hour).
+
+Our model learns quite fastly, and after 25 iterations, the training process
```
Comment (mhugo) [inline articles/predict_bike_availability.org]: fast

File: articles/predict_bike_availability.org [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+their cluster. Here we are in a regression problem, we are more focused on the
+level of bike availability, seen as a percentage.
+
+The following map shows such a quantity, for all Lyon stations:
+
+#+CAPTION: True shared bike availability, per station (red: empty station, blue: full station)
+#+NAME: fig:lyon_ground_truth_availability
+[[../image/lyon_groundtruth.png]]
+
+We have to compare it with the prediction provided by XGBoost model below. With
```
Comment (mhugo) [inline articles/predict_bike_availability.org:102]: the XGBoost model ?

**Review by mhugo — APPROVED** [review]

File: articles/cluster_stations.org (hunk 1 above)
Comment (delhomer) [reply articles/cluster_stations.org]: I believed that data was unvariable...

File: articles/cluster_stations.org (hunk 4 above)
Comment (garaud) [reply articles/cluster_stations.org:98]: I think 'Understand' is good.

**Pull request merged**